sdk/testing/src/main/java/io/opentelemetry/sdk/testing/context/StrictContextStorage.java

```diff
@@ -111,8 +116,9 @@ public Scope attach(Context context) {
 
     stackTrace = Arrays.copyOfRange(stackTrace, from, stackTrace.length);
     caller.setStackTrace(stackTrace);
+    pendingScopes.put(scope, caller);
```

Comment:
why not have this done in the StrictScope constructor, so it controls the full lifecycle, rather having it put in here, but removed internal to the `StrictScope.close()`?